src/cdomains/intDomain.ml

```diff
@@ -759,7 +772,13 @@ struct
     | `Excluded (s,r) -> min_of_range r
     | `Bot -> None
 
-  let of_excl_list t l = `Excluded (List.fold_right S.add l (S.empty ()), size t)
+  (* calculates the minimal extension of range r to cover the exclusion set s *)
+  let extend_range r s = S.fold (fun i s -> R.join s (size @@ Size.min_for i)) s r
```

Comment:
Could this also produce ranges that are not in C such as `[-7,8]`?
If yes, do we want such ranges?

Reply:
~~~c 
int test = 3;
int top;
if(top) {
    test = 2;
 }

 if(test != 5) {
   int test3 = -test;
   // test3 → Not {-5, 0}([-7,8])
 }
~~~

Reply:
I guess this also happens in `join` and `meet`, so not only here. So I guess it's fine.

Reply:
> Could this also produce ranges that are not in C such as [-7,8]? If yes, do we want such ranges?

Yes, why not? You could also manage an interval for the actual values instead of the sizes, if you want it to be more precise.